### PR TITLE
GH-47 | Add AnimatedContent helper to TextFieldScope

### LINKED ISSUE #47: Support animation wrappers (Crossfade / AnimatedContent / AnimatedVisibility) inside TextFieldScope trailing slot
## Feature Description

Extend `TextFieldScope` so the trailing slot of `KPSingleLineOutlinedTextField` and `KPMultiLineOutlineTextField` can support animated transitions between states, for example fading between a search icon and a clear (X) icon.

Currently, the trailing slot enforces a strict constraint: exactly one child carrying either `TrailingIconLayoutId` or `TrailingTextLayoutId`. This guarantee prevents using Compose animation wrappers (`Crossfade`, `AnimatedContent`, `AnimatedVisibility`) in this slot.

The validator checks the outermost measurable. Animation wrappers introduce their own `Layout` node without a `layoutId`, causing validation to fail even when the properly-tagged child exists deeper in the tree.

## Problem/Use Case

A common UX pattern swaps the trailing icon based on input state:

- Empty: show search icon
- Non-empty: show clear (X) icon

Plain conditionals work but produce abrupt transitions. Wrapping the trailing icon in `Crossfade` / `AnimatedContent` / `AnimatedVisibility` to soften the swap crashes at runtime with:

`IllegalArgumentException: Trailing content should be created using TextFieldScope.TrailingIcon or TextFieldScope.TrailingText`

**Reproducer**

```kotlin
@Preview(showBackground = true)
@Composable
fun SearchFieldCrossfadeReproducer() = TrendyolTheme {
    var query by remember { mutableStateOf("") }
    KPSingleLineOutlinedTextField(
        modifier = Modifier.width(280.dp),
        style = KPOutlinedTextFieldStyle.Outlined,
        value = query,
        onValueChange = { query = it },
        label = "Search",
        trailingContent = {
            Crossfade(targetState = query.isNotEmpty()) { hasQuery ->
                TextFieldScope.Icon(
                    imageVector = if (hasQuery) KPIcons.Fill.Cancel else KPIcons.Fill.Search,
                    onClick = { if (hasQuery) query = "" },
                    size = KPIconSize.Small,
                )
            }
        },
    )
}
```

**Verified on HEAD**

Running the reproducer above in Preview throws (truncated to the relevant frames):

```
java.lang.IllegalArgumentException: Trailing content should be created using
    TextFieldScope.TrailingIcon or TextFieldScope.TrailingText
  at com.trendyol.design.core.inputfield.container.CheckTextFieldLayoutIdsKt
      $CheckTextFieldLayoutIds$1$1.measure-3p2s80s(CheckTextFieldLayoutIds.kt:42)
  ...
  at com.trendyol.design.core.inputfield.impl.KPOutlinedTextFieldMeasurePolicy
      .measure-3p2s80s(TrendyolOutlinedTextFieldImpl.kt:159)
```

**Root Cause**

`CheckTextFieldLayoutIds(..., isSingleChildRequired = true)` inspects the outermost measurable (see `CheckTextFieldLayoutIds.kt:42`). `Crossfade` has its own `Layout` node without `layoutId`, so validation fails despite the correctly-tagged `Icon` existing one level deeper.

## Proposed Solution

**Option A: Add an animated helper to `TextFieldScope` (smallest scope)**

Introduce a wrapper function that applies the required `layoutId` to the outer node, keeping the validator satisfied. Non-breaking, additive, covers icon-swap use cases.

**Option B: Relax validator logic**

Walk the measurable subtree until finding a child with `TrailingIconLayoutId` / `TrailingTextLayoutId`, rather than checking only direct children. Lets consumers use animation wrappers naturally but weakens the "exactly one direct child" guarantee.

**Option C: Generic `AnimatedContainer` helper**

Like Option A but generic, accepting arbitrary content for `Text` transitions too (e.g., "Done" / spinner / "Saved").

## API Design

**Option A:** consumer usage + the helper it relies on:

> ⚠️ **NOTE:** The `@Suppress` annotation below is purely a local reproducer hack to verify the approach compiles and runs end-to-end. **It is not part of the proposal.** In the real fix the helper lives inside `core.inputfield.container` (same package as `TrailingIconLayoutId`), so the qualified reference and the suppression both disappear.

You can paste the block below directly into `app/src/main/java/com/trendyol/design/ui/component/inputfield/SingleLineTextField.kt` to try it locally.

```kotlin
// --- Consumer-side usage -------------------------------------------------

// Name follows this file's `Input_Field_1_SingleLine_*` preview pattern.
@Preview
@Composable
fun Input_Field_1_SingleLine_5_AnimatedTrailingIcon() {
    TrendyolTheme {
        var query by remember { mutableStateOf("") }
        KPSingleLineOutlinedTextField(
            style = KPOutlinedTextFieldStyle.Outlined,
            value = query,
            onValueChange = { query = it },
            label = "Search",
            trailingContent = {
                this@KPSingleLineOutlinedTextField.AnimatedIcon(
                    targetImageVector = if (query.isNotEmpty()) KPIcons.Fill.Cancel else KPIcons.Fill.Search,
                    onClick = { if (query.isNotEmpty()) query = "" },
                    size = KPIconSize.Small,
                    // Long fade so the animation is obvious in Preview.
                    animationSpec = tween(durationMillis = 1000)
                )
            },
        )
    }
}

// --- Helper --------------------------------------------------------------

@Composable
fun TextFieldScope.AnimatedIcon(
    targetImageVector: ImageVector,
    modifier: Modifier = Modifier,
    onClick: () -> Unit,
    size: IconSize = KPIconSize.Small,
    tint: Color = Color.Unspecified,
    animationSpec: FiniteAnimationSpec<Float> = tween(),
) {
    Box(
        modifier
            // key: layoutId on OUTER node so the trailing-slot validator accepts it.
            // `TrailingIconLayoutId` is `internal` to `core.inputfield.container`; the
            // @Suppress is a reproducer-only workaround. Once this helper is moved
            // next to `TextFieldScope` (same package), the qualified reference and
            // the suppression both disappear.
            .layoutId(
                @Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
                com.trendyol.design.core.inputfield.container.TrailingIconLayoutId
            )
            .size(size.dp)
            .clickable(onClick = onClick),
    ) {
        Crossfade(
            targetState = targetImageVector,
            animationSpec = animationSpec,
        ) { vector ->
            KPIcon(
                imageVector = vector,
                size = KPIconSize.Medium,
                modifier = Modifier
                    .align(Alignment.Center)
                    .padding(2.dp),
                tint = tint,
            )
        }
    }
}
```

## Implementation Details

- **Performance:** `Crossfade` / `AnimatedContent` run on the main animation clock; no extra cost beyond consumer implementation.
- **Backward compatibility:** Options A and C are purely additive. Option B loosens an internal invariant but preserves call-site compatibility.
- **Accessibility:** No change; underlying `KPIcon` retains semantics; only visual transition animates.
- **Testing strategy:** Add a test reproducing the crash, then verify the new helper renders without validation failure and animates correctly.
- **Public API:** Options A and C add new symbols. Option B changes only internal `CheckTextFieldLayoutIds` behavior.
- **Side note (separate minor bug):** The error message references `TextFieldScope.TrailingIcon` / `TextFieldScope.TrailingText`, but the actual scope functions are named `Icon` / `Text` (no "Trailing" prefix). Happy to fold the one-line string fix into the same PR.

## Examples/References

- Material 3 `TextField` accepts arbitrary composables in `trailingIcon`; consumers routinely wrap them with `AnimatedContent` / `Crossfade`.
- Jetpack Compose `AnimatedContent` documentation demonstrates the exact icon-swap pattern motivating this request.

## Priority/Impact

- [ ] Critical - Blocking current development
- [ ] High - Would significantly improve developer experience
- [x] Medium - Nice to have improvement
- [ ] Low - Minor enhancement

## Checklist

- [x] I have searched existing issues to ensure this is not a duplicate
- [x] I have provided clear use cases and examples
- [x] I have considered the impact on existing components
- [x] I have included design specifications or references

### PR DESCRIPTION
**Pull request by AhmetSIRIM**
### What does this PR do?

Adds `TextFieldScope.AnimatedContent<T>`, a generic helper that lets the trailing slot of `KPSingleLineOutlinedTextField` host animated transitions between `TextFieldScope.Icon` / `TextFieldScope.Text` content. Replaces the runtime crash that previously occurred when `Crossfade` / `AnimatedContent` / `AnimatedVisibility` were used directly inside `trailingContent`.

Implementation follows Option C as agreed on the issue: a generic wrapper that scales to future scope members, with an inner validator that enforces `TextFieldScope` elements inside the lambda but does not require a single child (so an `if` without `else` is valid for cases where one branch should render nothing).

## 📋 Related Issue

Closes #47

### How has this been tested?

- [x] Screenshot tests
- [ ] UI tests
- [x] Manual testing

`./gradlew check` passes locally (Detekt, apiCheck, lint, JVM unit tests). Screenshot baseline recorded with `./gradlew executeScreenshotTests -Precord` on Pixel 7 (AVD, API 33). Showkase preview `Input_Field_1_SingleLine_5_AnimatedTrailing` was verified locally for all demo states.

### Test Coverage

- [x] Main functionality
- [x] Edge cases
- [ ] Different Android versions
- [ ] Different screen sizes/densities
- [ ] RTL layout support
- [ ] Accessibility features

`kpSingleLineOutlinedTextFieldAnimatedTrailingContentTest` covers five trailing-content states: Empty/Filled icon swap, Empty/Filled with empty `else`, and Filled with Icon / Text mix. RTL was not specifically tested; the helper itself is direction-agnostic because `transitionSpec` is caller-controlled, so existing TextField RTL behavior propagates.

### Screenshots/Videos

Showkase preview `Input_Field_1_SingleLine_5_AnimatedTrailing` demonstrates fade, Icon / Text mix, empty-else, horizontal slide, and scale variants. Screenshot test baseline added under `app/screenshots/debug/`.

### General

- [x] Code follows project coding standards
- [x] Self-review completed
- [x] No debugging code left behind
- [x] Documentation updated if needed

### Testing

- [x] Existing tests still pass
- [x] New tests added if applicable
- [x] Screenshot tests updated if UI changed
- [x] Edge cases covered

### Breaking Changes

- [ ] This change introduces breaking changes
- [x] This change is backward compatible

Strictly additive. New public symbol on `TextFieldScope`; new internal `TrailingAnimatedContentLayoutId`; existing trailing-slot validator gains an extra entry in its allow-list. The error message wording was tightened to reference the actual scope members (`Icon` / `Text` / `AnimatedContent`) instead of the non-existent `TrailingIcon` / `TrailingText` names. This is the side note from the issue body, folded in as agreed.

## 🔄 Migration Required

- [x] No migration required
- [ ] Migration required (explain below)

**Migration Details:**

None. Existing call sites of `KPSingleLineOutlinedTextField` continue to work unchanged. `TextFieldScope.AnimatedContent` is purely opt-in.

## 📸 Screenshot Test Changes

- [ ] Screenshot tests were not affected
- [ ] Screenshot tests were updated
- [x] New screenshot tests were added

**Screenshot Changes Explained:**

A new test `kpSingleLineOutlinedTextFieldAnimatedTrailingContentTest` was added under `app/src/androidTest/java/inputfield/KPSingleLineOutlinedTextFieldTest.kt`, with five containers covering the supported trailing-content states. One new baseline PNG is committed under `app/screenshots/debug/`. No existing screenshots were modified.

---

## 🐛 Bug Fix Details

Not applicable. This PR is a feature addition (new opt-in helper); see Issue #47 for the underlying validator constraint that motivated the new surface.

---

### Components/Screens Covered

- [x] `KPSingleLineOutlinedTextField` trailing slot with `TextFieldScope.AnimatedContent`

### Test Scenarios Added

- [x] Default state
- [ ] Loading state
- [ ] Error state
- [ ] Success state
- [x] Empty state
- [ ] Focused state
- [ ] Disabled state
- [x] Different content lengths
- [ ] Interaction states

Specifically: empty-input icon, filled-input icon swap, empty `else` branch when empty, empty `else` branch when filled, and Icon / Text mix.

### Screenshot Test Coverage

- [ ] Different screen densities (mdpi, hdpi, xhdpi, xxhdpi)
- [ ] Different screen sizes (phone, tablet)
- [ ] RTL layout
- [ ] Different font scales
- [ ] Accessibility features enabled

Recorded against the project's standard Pixel 7 AVD baseline only, matching the existing screenshot test pattern in `KPSingleLineOutlinedTextFieldTest`.

### Screenshot Test Commands

```bash
# Generate new screenshots
./gradlew executeScreenshotTests -Precord

# Verify screenshots
./gradlew executeScreenshotTests
```

### Screenshot Quality Checklist

- [x] Screenshots are clear and properly framed
- [x] All relevant UI elements are visible
- [x] Screenshots represent realistic usage scenarios
- [x] Consistent with existing screenshot style
- [x] No unnecessary screenshot files added

### Coverage Improvement

- [x] Adds coverage for previously untested component
- [ ] Improves coverage for existing component
- [ ] Adds regression prevention for specific bug
- [x] Covers new component states/variations

---

## 📝 Additional Notes

- The KDoc on `TextFieldScope.AnimatedContent` documents every parameter (`targetState`, `modifier`, `transitionSpec`, `contentAlignment`, `label`, `content`) per the project's KDoc guidelines.
- The Showkase demo file uses two private helper functions (`horizontalSlideTransition`, `scaleAndFadeTransition`) and three duration constants (`FADE_DURATION_MS`, `SLIDE_DURATION_MS`, `SCALE_DURATION_MS`) to keep transitions readable and avoid Detekt MagicNumber / Wrapping findings.
- `core/api/core.api` was regenerated via `./gradlew :core:apiDump` to register the new public `AnimatedContent` member.

### CODE REVIEW

File: core/src/main/java/com/trendyol/design/core/inputfield/container/TextFieldScope.kt [hunk 1]
```diff
@@ -83,7 +92,63 @@ public object TextFieldScope {
...
+     * other content fails validation at runtime.
+     */
+    @Composable
+    public fun <T> AnimatedContent(
+        targetState: T,
+        modifier: Modifier = Modifier,
+        transitionSpec: AnimatedContentTransitionScope<T>.() -> ContentTransform = { fadeIn() togetherWith fadeOut() },
+        contentAlignment: Alignment = Alignment.Center,
+        label: String = "TextFieldScopeAnimatedContent",
+        content: @Composable TextFieldScope.(targetState: T) -> Unit,
```
Comment (turkergoksu) [inline core/src/main/java/com/trendyol/design/core/inputfield/container/TextFieldScope.kt:138]: Kinda nitpick but we can also add `contentKey` parameter to mimic Compose's AnimatedContent API.

Comment (AhmetSIRIM) [reply core/src/main/java/com/trendyol/design/core/inputfield/container/TextFieldScope.kt:138]: Good point, added in d0f50a5. Default `{ it }` to mirror Compose's AnimatedContent.

**Review by turkergoksu — DISMISSED** [review]
Thank you for the contribution 👑

**Review by turkergoksu — APPROVED** [review]

**Review by BurkClik — APPROVED** [review]
